es.go

```diff
@@ -1174,6 +1174,34 @@ func (c *Client) SnapshotAllIndices(repository string, snapshot string) error {
 	return err
 }
 
+func (c *Client) SnapshotAllIndicesWithBodyParams(repository string, snapshot string, bodyParams map[string]interface{}) error {
+	if repository == "" {
+		return errors.New("empty string for repository is not allowed")
+	}
+
+	if snapshot == "" {
+		return errors.New("empty string for snapshot is not allowed")
+	}
+
+	if bodyParams == nil {
```

Comment:
I think bodyParams should be allowed to be `nil`. Simply don't add any params in that case.

Reply:
Should we though? I mean there is a method to not use body params... I think that should be preferred if no params are required.

Reply:
hmmm I feel like maybe we should just depricate the previous one eventually and just have this use either bodyParams or not depending on if it is passed in or not